std/format.d

```diff
@@ -157,6 +157,11 @@ $(I Width):
     $(I empty)
     $(I Integer)
     $(B '*')
+$(I Underscore):
+    $(I empty)
+    $(B '_')
+    $(B '_') $(I Integer)
+    $(B '_*')
```

Comment:
The given grammar is just a suggestion. The parser actually allows you to give multiple order of width, precision and underscore. To fix that fact, I added another assert at the last unittest.

Thanks for the underscore tip.